src/ResourceManager/OperationalInsights/Commands.OperationalInsights/Query/InvokeOperationalInsightsQuery.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.OperationalInsights.Models;
+using Microsoft.Azure.OperationalInsights;
+using Microsoft.Rest;
+using System;
+using System.Management.Automation;
+using System.Net;
+
+namespace Microsoft.Azure.Commands.OperationalInsights.Query
+{
+    [Cmdlet("Invoke", "AzureRmOperationalInsightsQuery"), OutputType(typeof(PSQueryResponse))]
+    public class InvokeOperationalInsightsQuery : ResourceManager.Common.AzureRMCmdlet
+    {
+        [Parameter(Mandatory = true, HelpMessage = "The workspace ID.")]
+        public string WorkspaceId { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The query to execute.")]
+        public string Query { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The timespan to bound the query by.")]
+        public TimeSpan? Timespan { get; set; } = null;
+
+        [Parameter(Mandatory = false, HelpMessage = "Puts an upper bound on the amount of time the server will spend processing the query. See: https://dev.loganalytics.io/documentation/Using-the-API/Timeouts")]
+        [ValidateRange(1, int.MaxValue)]
+        public int Wait { get; set; } = int.MinValue;
+
+        [Parameter(Mandatory = false, HelpMessage = "If specified, rendering information for metric queries will be included in the response.")]
+        public SwitchParameter IncludeRender { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "If specified, query statistics will be included in the response.")]
+        public SwitchParameter IncludeStatistics { get; set; }
+
+        [Parameter(Mandatory = false,
```

Comment:
So when exactly is the parameter used? Does the user have a "DEMO_WORKSPACE"?

Reply:
So our API has a global workspace ID called "DEMO_WORKSPACE" that any customer can use as a sandbox/playground. The idea is to require the user to explicitly indicate they want to use the demo workspace here. I suppose that we could omit the switch and use the demo key anytime the workspace ID == "DEMO_WORKSPACE" if preferred.

Reply:
Let's talk about this more in the cmdlet review - I think the idea that the key would be automatically used is better than having a parameter that might confuse users.

Reply:
Sure - sounds good.

Reply:
Yes, this is preferred, please make this change